### PR TITLE
reload form and pass request vars on validation errors

### PR DESCRIPTION
**Pull request by KaiNissen**
[phab:T130360](https://phabricator.wikimedia.org/T130360)

### CODE REVIEW

File: src/Factories/FunFunFactory.php [hunk 1]
```diff
@@ -791,4 +792,8 @@ public function newShowDonationConfirmationUseCase( string $accessToken ): ShowD
 		);
 	}
 
+	public function newViolationPresenter() {
```
Comment (JeroenDeDauw) [inline src/Factories/FunFunFactory.php]: Think DonationForm should be in the name here as well

File: src/UseCases/AddDonation/AddDonationResponse.php [hunk 2]
```diff
@@ -50,6 +50,13 @@ public static function newFailureResponse( array $errors ): self {
 	private function __construct() {
 	}
 
+	public static function newViolationResponse( array $errors, AddDonationRequest $request ) {
+		$response = new self();
+		$response->validationErrors = $errors;
+		$response->precedingRequest = $request;
```
Comment (JeroenDeDauw) [inline src/UseCases/AddDonation/AddDonationResponse.php]: field does not exist

File: src/UseCases/AddDonation/AddDonationResponse.php [hunk 3]
```diff
@@ -50,6 +50,13 @@ public static function newFailureResponse( array $errors ): self {
...
+		return $response;
+	}
```
Comment (JeroenDeDauw) [inline src/UseCases/AddDonation/AddDonationResponse.php]: I don't understand why you added this

**Comment by JeroenDeDauw** [comment]
Needs rebase

File: src/UseCases/AddDonation/AddDonationResponse.php (hunk 3 above)
Comment (KaiNissen) [inline src/UseCases/AddDonation/AddDonationResponse.php]: Meh, those are leftovers from an approach I dropped. Removed, rebased, repushed.

**Comment by JeroenDeDauw** [comment]
Follow mup: https://github.com/wmde/FundraisingFrontend/pull/303

File: tests/templates/DonationForm.html.twig [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+Salutation: {$ anrede $}
+Title: {$ titel $}
+Company: {$ firma $}
+First name: {$ vorname $}
+Last name: {$ nachname $}
+Street: {$ strasse $}
+Postal code: {$ plz $}
+City: {$ ort $}
+Country code: {$ country $}
+Email address: {$ email $}
```
Comment (JeroenDeDauw) [inline tests/templates/DonationForm.html.twig]: So this is a placeholder form, that should still be replaced by some real frontend stuffs? How would that work?

File: src/Presentation/Presenters/DonationFormViolationPresenter.php [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+ */
+class DonationFormViolationPresenter {
+
+	private $template;
+
+	public function __construct( TwigTemplate $template ) {
+		$this->template = $template;
+	}
+
+	public function present( AddDonationRequest $request ): string {
```
Comment (JeroenDeDauw) [inline src/Presentation/Presenters/DonationFormViolationPresenter.php]: So the errors themselves are not shown?

File: tests/templates/DonationForm.html.twig (hunk 4 above)
Comment (KaiNissen) [inline tests/templates/DonationForm.html.twig]: This is for the system test to make sure all required data is passed to the presenter. The actual form will use the parameters to populate/restore the form fields with the given values so users don't have to fill the form again.

**Pull request merged**